### PR TITLE
GDPR DPIA [WIP]

### PR DESCRIPTION
**Pull request by cloudify**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: gdpr/DPIA.md [hunk 1]
```diff
@@ -0,0 +1,885 @@
...
+accesso viene richiesto all'utente di impostare un codice PIN (ed opzionalmente
+di configurare un codice di accesso biometrico se il dispositivo lo consente).
+
+Nel caso in cui la sessione di autenticazione verso il backend dell'app è
+ancora valida[^validita-sessione-backend], il codice PIN verrà richiesto
+all'utente per sbloccare l'applicazione quando questa esce dal background.
+
+[^validita-sessione-backend]: Attualmente la sessione viene considerata valida per un tempo di 30 giorni.
+
+##### Autenticazione verso il backend dell'app
```
Comment (cloudify) [inline gdpr/DPIA.md]: @fpersico @lussoluca date un occhiata qui

File: gdpr/DPIA.md [hunk 2]
```diff
@@ -0,0 +1,885 @@
...
+
+L'app mantiene una sessione verso il backend creata al completamento con
+successo dell'autenticazione SPID ma scollegata dalla sessione SPID che
+l'utente stabilisce con l'Identity Provider.
+
+Il flusso di autenticazione dell'app con il backend tramite la sessione è il
+seguente:
+
+1. L'utente inizia il processo di autenticazione SPID dall'interno dell'app.
+1. Il backend genera un _token_ di sessione (una stringa alfanumerica casuale)
```
Comment (lussoluca) [inline gdpr/DPIA.md]: per essere precisi il token viene generato quando il backend riceve indietro l'asserzione SPID

File: gdpr/DPIA.md [hunk 3]
```diff
@@ -0,0 +1,885 @@
...
+   scelto dall'utente, associandolo al _token_ generato (Figura \vref{figura-flusso-autenticazione-app}).
+1. Al termine del flusso di autenticazione SPID, il backend riceve l'asserzione
+   SAML firmata dall'IdP e contente gli attributi SPID richiesti.
+1. Gli attributi richiesti vengono salvati in un database locale ed associati
+   al _token_ di sessione condiviso con l'app.
+1. L'app effettua una chiamata alle API del backend usando il _token_ di
+   sessione come meccanismo di autenticazione _bearer token_.
+1. Il backend recupera gli attributi associati al _token_ ed esegue l'operazione
+   richiesta, associandola all'utente SPID.
+1. Se il _token_ risulta creato da oltre 30 giorni, il backend risponderà
```
Comment (lussoluca) [inline gdpr/DPIA.md]: la durata del token è configurabile con una variabile d'ambiente

**Pull request merged**